lib/index.js

```diff
@@ -122,149 +229,23 @@ var util = require('./common/util.js');
  * //-
  */
 function gcloud(config) {
-  return {
-    bigquery: function(options) {
-      options = options || {};
-      return new BigQuery(util.extendGlobalConfig(config, options));
-    },
-    datastore: new Datastore(config),
-    dns: function(options) {
-      options = options || {};
-      return new DNS(util.extendGlobalConfig(config, options));
-    },
-    pubsub: function(options) {
-      options = options || {};
-      return new PubSub(util.extendGlobalConfig(config, options));
-    },
-    search: function(options) {
-      options = options || {};
-      return new Search(util.extendGlobalConfig(config, options));
-    },
-    storage: function(options) {
-      options = options || {};
-      return new Storage(util.extendGlobalConfig(config, options));
+  return Object.keys(apis).reduce(function(gcloudExposedApi, apiName) {
+    var Class = apis[apiName];
+
+    if (scopedApis[apiName]) {
+      gcloudExposedApi[apiName] = new Class(config);
+    } else {
+      gcloudExposedApi[apiName] = function(options) {
+        return new Class(util.extendGlobalConfig(config, options));
+      };
```

Comment:
We're using reduce to iterate over 6 APIs and creating a map to construct one of those APIs in a different fashion. Considering how few APIs we actual offer at the moment, this feels sort of over the top. I'm not opposed to it, eventually if we offer more APIs (10+) it'll make more sense, but given how many we offer now it just feels like overkill to me!